### PR TITLE
SAK-45983 move the code that creates a content review assignment into service

### PR DESCRIPTION
**Pull request by ottenhoff**
This allows the import from site code to properly create a new content-review assignment.

### CODE REVIEW

**Comment by ottenhoff** [comment]
Alternative approach to all of this refactoring would be to prevent creation of a published assignment during import .... force it to be a draft if the old assignment used content review .....

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 1]
```diff
@@ -4070,6 +4146,17 @@ public void updateEntityReferences(String toContext, Map<String, String> transve
...
                     updateAssignment(nAssignment);
 
+                    // review service
+                    if (oAssignment.getContentReview()) {
+                        nAssignment.setContentReview(true);
+                        String errorMsg = createContentReviewAssignment(nAssignment, nAssignmentRef, nAssignment.getOpenDate(), nAssignment.getDueDate(), nAssignment.getCloseDate());
+                        if (StringUtils.isNotBlank(errorMsg)) {
+                            log.warn("Error while copying old assignments and creating content review link: {}", errorMsg);
+                            nAssignment.setDraft(true);
+                            updateAssignment(nAssignment);
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4156]: Maybe updateAssignment should be called once here

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java [hunk 2]
```diff
@@ -3749,6 +3751,84 @@ public void postReviewableSubmissionAttachments(AssignmentSubmission submission)
...
+        opts.put("title", assignment.getTitle());
+        opts.put("instructions", assignment.getInstructions());
+        if (!assignment.getAttachments().isEmpty()) {
+            opts.put("attachments", new ArrayList<>(assignment.getAttachments()));
+        }
+        try {
+            contentReviewService.createAssignment(assignment.getContext(), assignmentRef, opts);
+            return "";
+        } catch (Exception e) {
+            log.error(e.getMessage());
```
Comment (ern) [inline assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:3827]:
```suggestion
            log.error(e.toString());
```

**Review by ern — CHANGES_REQUESTED** [review]
Couple of minor suggestions

File: assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java (hunk 1 above)
Comment (ottenhoff) [reply assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java:4156]: I am just copying the same pattern used when instructor creates assignment from scratch

**Review by ern — APPROVED** [review]

**Pull request merged**